torchinductor/codegen/triton.py

```diff
@@ -611,7 +611,10 @@ def store(self, name, index, value, mode=None):
         index, mask = self.indexing(index, value)
         if mode is None:
             line = f"tl.store({var} + {index}, {value}, {mask})"
-        elif mode == "atomic_add":
+        elif mode == "atomic_add" or mode == "atomic_add_exclude_self":
+            if mode == "atomic_add_exclude_self":
```

Comment:
I don't exactly understand this one...  if we zero it out then why do we need the atomic_add? Could this just be a normal store?
```
x=0
x+=y
```
seems the same as
```
x=y
```

Reply:
`atomic_add` is needed because of the `index` in `scatter_add` can make it add to the same element multiple times, e.g.
```
>>> torch.scatter_add(torch.zeros(2, 2), 0, torch.tensor([[1],[1]]), torch.ones(2,2))
tensor([[0., 0.],
        [2., 0.]]
```

So even we zero out the destination, we still need an `atomic_add` here.

Reply:
how do we know the zeroing happens before the adding?

Reply:
Oh right, good catch, all zeroing will be racing.